src/network.js

```diff
@@ -384,182 +365,173 @@ Network.prototype = {
+        activation += `${data.activate[neuron][sentence].join('')}\n`;
     }
 
     // build outputs
     activation += "var output = [];\n";
     for (var i in data.outputs)
-      activation += "output[" + i + "] = F[" + data.outputs[i] + "];\n";
+      activation += `output[${i}] = F[${data.outputs[i]}];\n`;
     activation += "return output;\n}";
 
     // reference all the positions in memory
-    var memory = activation.match(/F\[(\d+)\]/g);
-    var dimension = 0;
-    var ids = {};
-    for (var address in memory) {
-      var tmp = memory[address].match(/\d+/)[0];
+    const memory = activation.match(/F\[(\d+)\]/g);
+    let dimension = 0;
+    const ids = {};
+    for (let address in memory) {
+      const tmp = memory[address].match(/\d+/)[0];
       if (!(tmp in ids)) {
         ids[tmp] = dimension++;
       }
     }
-    var hardcode = "F = {\n";
+    let hardcode = "F = {\n";
     for (var i in ids)
-      hardcode += ids[i] + ": " + this.optimized.memory[i] + ",\n";
-    hardcode = hardcode.substring(0, hardcode.length - 2) + "\n};\n";
-    hardcode = "var run = " + activation.replace(/F\[(\d+)]/g, function(
-      index) {
-      return 'F[' + ids[index.match(/\d+/)[0]] + ']'
-    }).replace("{\n", "{\n" + hardcode + "") + ";\n";
+      hardcode += `${ids[i]}: ${this.optimized.memory[i]},\n`;
+    hardcode = `${hardcode.substring(0, hardcode.length - 2)}\n};\n`;
+    hardcode = `var run = ${activation.replace(/F\[(\d+)]/g, index => 'F[' + ids[index.match(/\d+/)[0]] + ']').replace("{\n", "{\n" + hardcode + "")};\n`;
     hardcode += "return run";
 
     // return standalone function
     return new Function(hardcode)();
-  },
-
+  }
 
   // Return a HTML5 WebWorker specialized on training the network stored in `memory`.
   // Train based on the given dataSet and options.
   // The worker returns the updated `memory` when done.
-  worker: function(memory, set, options) {
+  worker(memory, set, options) {
 
     // Copy the options and set defaults (options might be different for each worker)
-    var workerOptions = {};
-    if(options) workerOptions = options;
+    let workerOptions = {};
+    if (options) workerOptions = options;
     workerOptions.rate = options.rate || .2;
     workerOptions.iterations = options.iterations || 100000;
     workerOptions.error = options.error || .005;
     workerOptions.cost = options.cost || null;
     workerOptions.crossValidate = options.crossValidate || null;
 
     // Cost function might be different for each worker
-    costFunction = "var cost = " + (options && options.cost || this.cost || Trainer.cost.MSE) + ";\n";
-    var workerFunction = Network.getWorkerSharedFunctions();
+    costFunction = `var cost = ${options && options.cost || this.cost || Trainer.cost.MSE};\n`;
+    let workerFunction = Network.getWorkerSharedFunctions();
     workerFunction = workerFunction.replace(/var cost = options && options\.cost \|\| this\.cost \|\| Trainer\.cost\.MSE;/g, costFunction);
```

Comment:
That would be perfect, the way it is done now is just a hack. But we need to have a way to train without blocking the UI on the browser.